server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -1064,7 +1064,13 @@ public void run() {
 
   public static void main(String[] args) throws Exception {
     try (Compactor compactor = new Compactor(new ConfigOpts(), args)) {
-      compactor.runServer();
+      try {
+        compactor.runServer();
+      } catch (Exception e) {
+        System.err.println("Compactor died, exception thrown from runServer.");
+        e.printStackTrace();
+        LOG.error("Compactor died, exception thrown from runServer.", e);
+      }
```

Comment:
This is fine but causes it to exit with a normal exit code, 0. Normally an exception being thrown through the main class would cause a non-zero exit code.